lib/k8s/client/mint/http_adapter.ex

```diff
@@ -148,20 +148,30 @@ defmodule K8s.Client.Mint.HTTPAdapter do
   end
 
   @impl true
-
   def handle_call({:request, method, path, headers, body, pool, stream_to}, from, state) do
     caller_ref = from |> elem(0) |> Process.monitor()
 
-    case Mint.HTTP.request(state.conn, method, path, headers, body) do
-      {:ok, conn, request_ref} ->
-        state =
-          put_in(
-            state.requests[request_ref],
-            Request.new(pool: pool, stream_to: stream_to, caller_ref: caller_ref)
-          )
-
-        {:reply, {:ok, request_ref}, struct!(state, conn: conn)}
-
+    # stream body if there is one.
+    {body, pending_request_body} =
+      case body do
+        nil -> {body, body}
+        body -> {:stream, String.to_charlist(body)}
```

Comment:
We don't need to stream if the entire request fits in the window. I'm pretty sure that saves a round trip, so might be a nice optimization to put as a TODO.

Reply:
Indeed, nor if it's HTTP1. 
The HTTP1 case I'll do now. The other one probably a todo for now.

Reply:
actually, not I can add both checks - non need for the TODO